.editorconfig

```diff
@@ -1,134 +1,206 @@
+# editorconfig: http://editorconfig.org/
+# Help developers standardize spaces, tabs, encoding, end-line characters across editors
+
+# top-most .editorconfig file
 root = true
 
-# Rules in this file were initially inferred by Visual Studio IntelliCode from the S:\Repos\dep.controls codebase based on best match to current usage at 11/15/2018
-# You can modify the rules from these initially generated values to suit your own policies
-# You can learn more about editorconfig here: https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-code-style-settings-reference
-[*.cs]
+# defaults for all files
+[*]
+charset = utf-8
+end_of_line = crlf
+indent_size = 4
+indent_style = space
+insert_final_newline = true
+trim_trailing_whitespace = true
 
-#Core editorconfig formatting - indentation
+# markdown, diff overrides
+# two trailing spaces are required for <br/> and hard line-breaks in markdown files
+# see: (https://daringfireball.net/projects/markdown/syntax#p) and (http://spec.commonmark.org/0.27/#hard-line-break)
+[*.{md,diff}]
+trim_trailing_whitespace = false
 
-#use soft tabs (spaces) for indentation
-indent_style = space
-indent_size = 4
+[*.{md,xml,gprops,man,natvis}]
+indent_size = 2
+
+# manifest validation tool requires BOM
+[*.man]
 charset = utf-8-bom
-end_of_line = crlf
 
-#Formatting - indentation options
+# XML-based MSBuild and Visual Studio files
+[*.{props,targets,settings,*proj,vcxitems,filters}]
+indent_size = 2
 
-#indent switch case contents.
-csharp_indent_case_contents = true
-#csharp_indent_case_contents_when_block
-csharp_indent_case_contents_when_block = true
-#indent switch labels
-csharp_indent_switch_labels = true
+# Exceptions to the above *proj wildcard
+[*.vdproj]
+indent_size = 4
 
-#Formatting - new line options
+# Visual Studio uses hard tabs for SLN files, so don't fight it
+[*.sln]
+indent_style = tab
 
-#place catch statements on a new line
-csharp_new_line_before_catch = true
-#place else statements on a new line
-csharp_new_line_before_else = true
-#require braces to be on a new line for accessors, methods, object_collection, control_blocks, types, properties, and lambdas (also known as "Allman" style)
-csharp_new_line_before_open_brace = accessors, methods, object_collection, control_blocks, types, properties, lambdas
+# YAML overrides
+[*.{yml,yaml}]
+indent_size = 2
 
-#Formatting - organize using options
+# package.json overrides
+# Updating package.json with NPM will revert indentation to 2 spaces so to
+# reduce churn, let's align to NPM and specify indent size 2
+[package.json]
+indent_size = 2
 
-#do not place System.* using directives before other using directives
-dotnet_sort_system_directives_first = false
+# JSON formats
+[{*.json,*.testlist,testmd.definition}]
+indent_size = 2
 
-#Formatting - spacing options
+[*.{rc}]
+trim_trailing_whitespace = true
+insert_final_newline = true
```

Comment:
Line 56+57 are dupes of line 13+14 in the `[*]` section. Does this act is an _override_ to `[*]` definitions or is `[*]` a peer only used when a matching extension's not otherwise found?

If these are peers a lot of extensions are missing values.

If these are overrides to `[*]` then there's a lot of redundancies in the file e.g. delete lines 35, 56+57, 60+61, 64+65 and probably a few others.

Reply:
Per the [docs](https://editorconfig-specification.readthedocs.io/#id6) the rules are additive/overrides:

>Files are read top to bottom and the most recent rules found take precedence. If multiple EditorConfig files have matching sections, the rules from the closer EditorConfig file are read last, so pairs in closer files take precedence.

So anything defined for [*] at the top doesn't need to be redundantly defined later

Reply:
I want to keep this in sync with the os.2020 version.
You are correct that this is not needed but being explicit and not depending on the other declarations argues for keeping them. Let me know if you strongly disagree.

Reply:
Are you thinking we should keep this and os.2020 copies in sync? Changes here port to there and vice versa?

>being explicit and not depending on the other declarations argues for keeping them

Problem is it's _half_ explicit. If you really think explicit is good then lines 9-14 should be in every section. If the additive/override model better then we should embrace that. The half-and-half is what I question. This is particularly important if you think WinAppSDK and os.2020 should be kept in sync, as that precludes WinAppSDK from fixing the half-and-half approach.

I prefer the override approach as the more manageable and maintainable solution. Less is more.

Thoughts?

Reply:
good point. I want to get this in for now to avoid wasting time in PRs. 

I filed http://task.ms/37331332 to go all in on override.